enclave_wrangler/objects_api.py

```diff
@@ -324,41 +324,77 @@ def update_db_with_new_objects(objects=None):
     pass
 
 
-def get_new_objects(since: datetime = None):
-    """Get new objects"""
-    # https://www.palantir.com/docs/foundry/api/ontology-resources/objects/object-basics/#filtering-objects
-    # https://unite.nih.gov/workspace/data-integration/restricted-view/preview/ri.gps.main.view.af03f7d1-958a-4303-81ac-519cfdc1dfb3
-    yesterday = (datetime.now() - timedelta(
-        days=1)).isoformat() + 'Z'  # works: 2023-01-01T00:00:00.000Z
-    since = since if since else yesterday
+# TODO: finish this
+def get_all_new_objects(since: Union[datetime, str]) -> Dict[str, List]:
+    """Get new objects needed to update database.
+
+    Resources:
+    https://www.palantir.com/docs/foundry/api/ontology-resources/objects/object-basics/#filtering-objects
+    https://unite.nih.gov/workspace/data-integration/restricted-view/preview/ri.gps.main.view.af03f7d1-958a-4303-81ac-519cfdc1dfb3
+    """
+    since = str(since)
+    csets_and_members: Dict[str, List] = get_new_cset_and_member_objects(since, return_type='flat')
+    # TODO:
+    researchers = {}
+    # TODO:
+    projects = {}
+    # TODO: what else?
+    pass
+    return {**csets_and_members, **researchers, **projects}
+
+
+# TODO
+#  - future: updates: if someone made a change to something we already fetched, how do we know?
+#  - how to refresh when user creates something?: (a) have the user refresh the db immediately after creating something,
+#    and just download everything since list 'since' date? Or (b) fetch only what they updated (if so, we can't do
+#    inserts; we have to do updates on IDs)
+def get_new_cset_and_member_objects(since: Union[datetime, str], return_type=['flat', 'hierarchical'][0]) -> Dict[str, List]:
```

Comment:
`get_new_cset_and_member_objects()`: Renamed this from 'get_new_objects()` and finished it up.